core/src/main/java/com/netflix/conductor/core/execution/mapper/DoWhileTaskMapper.java

```diff
@@ -0,0 +1,87 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.conductor.core.execution.mapper;
+
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.workflow.TaskType;
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.execution.SystemTaskType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An implementation of {@link TaskMapper} to map a {@link WorkflowTask} of type {@link TaskType#DO_WHILE}
+ * to a {@link Task} of type {@link SystemTaskType#DO_WHILE}
+ */
+public class DoWhileTaskMapper implements TaskMapper {
+
+    public static final Logger logger = LoggerFactory.getLogger(DoWhileTaskMapper.class);
+
+    /**
+     * This method maps {@link TaskMapper} to map a {@link WorkflowTask} of type {@link TaskType#DO_WHILE} to a {@link Task} of type {@link SystemTaskType#DO_WHILE}
+     * with a status of {@link Task.Status#IN_PROGRESS}
+     *
+     * @param taskMapperContext: A wrapper class containing the {@link WorkflowTask}, {@link WorkflowDef}, {@link Workflow} and a string representation of the TaskId
+     * @return: A {@link Task} of type {@link SystemTaskType#DO_WHILE} in a List
+     */
+    @Override
+    public List<Task> getMappedTasks(TaskMapperContext taskMapperContext) {
+
+        logger.debug("TaskMapperContext {} in DoWhileTaskMapper", taskMapperContext);
+
+        WorkflowTask taskToSchedule = taskMapperContext.getTaskToSchedule();
+        Workflow workflowInstance = taskMapperContext.getWorkflowInstance();
+
+        Task task = workflowInstance.getTaskByRefName(taskToSchedule.getTaskReferenceName());
+        if (task != null && task.getStatus().isTerminal()) {
+            //Since loopTask is already completed no need to schedule task again.
+            return Collections.emptyList();
+        }
+
+        String taskId = taskMapperContext.getTaskId();
+        List<Task> tasksToBeScheduled = new ArrayList<>();
+        int retryCount = taskMapperContext.getRetryCount();
+
+        Task loopTask = new Task();
+        loopTask.setTaskType(SystemTaskType.DO_WHILE.name());
+        loopTask.setTaskDefName(taskToSchedule.getName());
+        loopTask.setReferenceTaskName(taskToSchedule.getTaskReferenceName());
+        loopTask.setWorkflowInstanceId(workflowInstance.getWorkflowId());
+        loopTask.setCorrelationId(workflowInstance.getCorrelationId());
+        loopTask.setWorkflowType(workflowInstance.getWorkflowName());
+        loopTask.setScheduledTime(System.currentTimeMillis());
+        loopTask.setTaskId(taskId);
+        loopTask.setStatus(Task.Status.IN_PROGRESS);
```

Comment:
@manan164 If a loopLask is simple (worker) task , it will be first "Scheduled" and then marked "IN_PROGRESS" when a worker polls for it.